### DISCUSSION TITLE
Export a Custom Campaign, Test Cases, and All Notes?

### DISCUSSION THREAD

**Discussion by ghost** [SecurityRiskAdvisors/VECTR#272] [Q&A]
Hello! I wanted to see if there's a way to export a custom campaign I created in VECTR with all of the test cases and notes I made in the campaign? It'd be great to be able to export it if I wanted to share, archive, or even import the custom campaign into another VECTR instance. There doesn't appear to be option to export from the campaign dashboard:

![vectr](https://github.com/SecurityRiskAdvisors/VECTR/assets/121637879/adf971c2-afd6-4eba-9760-bcf995aae13b)

**Comment by thebleucheese** [comment]
> Hello! I wanted to see if there's a way to export a custom campaign I created in VECTR with all of the test cases and notes I made in the campaign? It'd be great to be able to export it if I wanted to share, archive, or even import the custom campaign into another VECTR instance. There doesn't appear to be option to export from the campaign dashboard:
> 

You can export an environment to CSV from the environment menu and the vectr-tools project (https://github.com/SecurityRiskAdvisors/vectr-tools) will allow you to do a rough lossy import via the API. 

Importing and exporting local data is very complex to handle in the UI. We've considered it and have had it as an internal feature request, but data collisions and existing data lookups are more difficult to handle than in the library. For now, I'd recommend the API for anything like this and that way you can handle any of your specific business needs in code.

**Reply by ghost** [reply]
Thanks! I'll give this a shot.

**Comment by ssnkhan** [comment]
Just posting here, the release of the VECTR Assessment Transfer tool, also developed by VECTR, which is a more elegant solution: https://github.com/SecurityRiskAdvisors/vat
